stdlib/universe/difference_test.go

```diff
@@ -723,3 +723,242 @@ func TestDifference_Process(t *testing.T) {
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(1), 20.0},
+					{execute.Time(2), 30.0},
+					{execute.Time(3), 50.0},
+					{execute.Time(4), 30.0},
+					{execute.Time(5), -40.0},
+				},
+			}},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(2), 10.0},
+					{execute.Time(3), 20.0},
+					{execute.Time(4), 30.0},
+					{execute.Time(5), nil},
+				},
+			}},
+		},
+		// keepFirst: true, initialZero: true
+		{
+			name: "float with KeepFirst and InitialZero",
+			spec: &universe.DifferenceProcedureSpec{
+				Columns:     []string{execute.DefaultValueColLabel},
+				NonNegative: true,
+				KeepFirst:   true,
+				InitialZero: true,
+			},
+			data: []flux.Table{&executetest.Table{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+					{Label: "t", Type: flux.TString},
+				},
+				Data: [][]interface{}{
+					{execute.Time(1), 20.0, "a"},
+					{execute.Time(2), 30.0, "b"},
+					{execute.Time(3), 50.0, "c"},
+					{execute.Time(4), 30.0, "d"},
+					{execute.Time(5), 40.0, "e"},
+				},
+			}},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+					{Label: "t", Type: flux.TString},
+				},
+				Data: [][]interface{}{
+					{execute.Time(1), float64(0), "a"},
+					{execute.Time(2), 10.0, "b"},
+					{execute.Time(3), 20.0, "c"},
+					{execute.Time(4), 30.0, "d"},
+					{execute.Time(5), 10.0, "e"},
+				},
+			}},
+		},
+		{
+			name: "with null non negative and InitialZero",
+			spec: &universe.DifferenceProcedureSpec{
+				Columns:     []string{"a", "b", "c"},
+				NonNegative: true,
+				InitialZero: true,
+			},
+			data: []flux.Table{&executetest.Table{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "a", Type: flux.TFloat},
+					{Label: "b", Type: flux.TFloat},
+					{Label: "c", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(0), nil, 1.0, 2.0},
+					{execute.Time(1), 6.0, 2.0, nil},
+					{execute.Time(2), 4.0, 2.0, 4.0},
+					{execute.Time(3), 10.0, 10.0, 2.0},
+					{execute.Time(4), nil, nil, 1.0},
+				},
+			}},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "a", Type: flux.TFloat},
+					{Label: "b", Type: flux.TFloat},
+					{Label: "c", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(1), nil, 1.0, nil},
+					{execute.Time(2), 4.0, 0.0, 2.0},
+					{execute.Time(3), 6.0, 8.0, 2.0},
+					{execute.Time(4), nil, nil, 1.0},
+				},
+			}},
+		},
+		{
+			name: "float with tags and InitialZero",
```

Comment:
This looks like the same test case as the one above it. It might be useful to have this same test case with `keepFirst: true`. Was that the intention or is there a difference I'm not seeing?